src/harness/reference_models/interference/interference.py

```diff
@@ -411,50 +412,82 @@ def computeInterferenceFssCochannel(cbsd_grant, constraint, fss_info, max_eirp):
-      constraint.low_frequency > cbsd_grant.high_frequency and
-      cbsd_grant.low_frequency > offset and
-             cbsd_grant.high_frequency > offset):
-    fss_mask_loss = linearToDb((cbsd_freq_range / MHZ) * 0.6)
+  # Define the initial attenuation values to 0
+  fss_mask_attenuation = 0
+  fss_mask_attenuation_seg1 = 0
+  fss_mask_attenuation_seg2 = 0
+  
+  # If the grant overlap frequency lies between fp1-50 MHz to fp1, where fp1 is the starting frequency of FSS Passband
+  # Segment 1 calculation is applied
+  if ( offset <= cbsd_grant.low_frequency and
+         constraint.high_frequency >= cbsd_grant.high_frequency):
+    # Calculate the total of 1MHZ frequencies present in the grant frequency range
+    num_one_mhz_bin_seg1 = int(cbsd_freq_range/MHZ)
+    # Calculate the fss_mask for every 1MHZ frequency
+    for i in range (0, num_one_mhz_bin_seg1):
+      fss_mask_attenuation +=dbToLinear(-(int((constraint.high_frequency - cbsd_grant.low_frequency)/MHZ) - 0.5 -i)*0.6 - 0.5)
+    # Calculate the average attenuation value per MHZ
+    fss_mask_attenuation_seg1 = fss_mask_attenuation/num_one_mhz_bin_seg1
+
+  # If the grant overlap frequency lies between fp1-150 MHz to fp1-50, where fp1 is the starting frequency of FSS Passband
+  # Segment 2 calculation is applied
+  elif ((offset - 100*MHZ) <= cbsd_grant.low_frequency and
+          offset >= cbsd_grant.high_frequency):
+    fss_mask_attenuation = 0
+    # Calculate the total of 1MHZ frequencies present in the grant frequency range
+    num_one_mhz_bin_seg2 = int(cbsd_freq_range/MHZ)
+    # Calculate the fss_mask for every 1MHZ frequency
+    for i in range (0, num_one_mhz_bin_seg2):
+      fss_mask_attenuation += dbToLinear(-(int((offset - cbsd_grant.low_frequency)/MHZ) - 0.5 -i)*0.25 - 30.5)
+    # Calculate the average attenuation value per MHZ
+    fss_mask_attenuation_seg2 = fss_mask_attenuation/num_one_mhz_bin_seg2
+
+  # If the grant frequency overlaps both the segments, the overlapping
+  # range is considered and calculation is done on both segments
+  elif ((offset - 100*MHZ) <= cbsd_grant.low_frequency and
+        constraint.high_frequency >= cbsd_grant.high_frequency):
+    # Get the total of 1MHZ frequencies overlapping within the segment1 range
+    num_one_mhz_bin_seg1 = int((cbsd_grant.high_frequency - offset)/MHZ)
+    fss_mask_attenuation = 0
+    for i in range (0, num_one_mhz_bin_seg1):
+      fss_mask_attenuation += dbToLinear((-(int((constraint.high_frequency - offset)/MHZ)) - 0.5 -i)*0.6 - 0.5)
+    fss_mask_attenuation_seg1 = fss_mask_attenuation/num_one_mhz_bin_seg1
+    fss_mask_attenuation = 0
+    # Get the total of 1MHZ frequencies overlapping within the segment2 range
+    num_one_mhz_bin_seg2 = int((offset - cbsd_grant.low_frequency)/MHZ)
+    for i in range (0, num_one_mhz_bin_seg2):
+      fss_mask_attenuation += dbToLinear((-(num_one_mhz_bin_seg2) - 0.5 -i)*0.25 - 30.5)
+    fss_mask_attenuation_seg2 = fss_mask_attenuation/num_one_mhz_bin_seg2
```

Comment:
This logic is wrong, you cannot add fss_mask_attenuation_seg1 and seg2 like that after having done the average. It needs to be added before the division.

This whole case needs can take the exampled of the first 2 cases. But instead of that, please check my last comment which fixes the whole function.

Reply:
Each segment's average is independent of other. The segments are broken and contribution is independently accounted.